src/fides/api/service/connectors/s3_connector.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.schemas.connection_configuration.connection_secrets_s3 import S3Schema
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.query_config import QueryConfig
+from fides.api.util.aws_util import get_aws_session
+from fides.api.util.collection_util import Row
+from fides.connectors.models import ConnectorFailureException
+
+
+class S3Connector(BaseConnector):
+    """
+    AWS S3 Connector - this is currently used just to test connections to S3.
+
+    NOTE: No DSR processing is yet supported for S3.
+    """
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for s3"""
+        config = S3Schema(**self.configuration.secrets or {})
+        return get_aws_session(
+            auth_method=config.auth_method.value,
+            storage_secrets=config.dict(),  # type: ignore
+            assume_role_arn=config.aws_assume_role_arn,
+        )
+
+    def query_config(self, node: ExecutionNode) -> QueryConfig[Any]:
+        """DSR execution not yet supported for S3"""
+        raise NotImplementedError()
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to AWS S3 and lists buckets to validate credentials.
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+        try:
+            session = self.client()
+            s3 = session.resource("s3")
+            s3.buckets.all()
+        except Exception as error:
+            raise ConnectorFailureException(str(error))
```

Comment:
I'd raise a ConnectionException instead - the endpoint that tests connections is looking for this exception. Otherwise, you'll have unhandled failures that will return a 500.  (Also looks like the UI shows green in that case), separate issue.

Reply:
thanks for flagging that!

Reply:
adjusted in https://github.com/ethyca/fides/pull/4930/commits/1cfd96f4d75e57a8d28263c0bf4a49fdb1080022